app/controllers/api/automate_workspaces_controller.rb

```diff
@@ -1,8 +1,41 @@
 module Api
   class AutomateWorkspacesController < BaseController
     def edit_resource(type, id, data = {})
+      raise BadRequestError, "Must specify an id for starting a #{type} resource" unless id
+
       obj = resource_search(id, type, collection_class(type))
       obj.merge_output!(data)
     end
+
+    def decrypt_resource(type, id = nil, data = nil)
+      raise BadRequestError, "Must specify an id for starting a #{type} resource" unless id
+
+      obj = resource_search(id, type, collection_class(type))
+      data['resources'] ? decrypt_all(obj, data) : decrypt_one(obj, data)
```

Comment:
Is this necessary? IIRC the update code should call this for each set of data in the "resources" node, so this should never have to handle the `decrypt_all` case

Reply:
@imtayadeway 
These resources are not like vmdb objects, these are embedded objects in the AutomateWorkspace input column. Which stores a Hash with the objects and attributes.

Reply:
I get that part - what I don't understand is `data` should represent the thing *under* "resources", so I don't understand why you're expecting that here.

Reply:
I don't think our API manager supports "resources" other than collection or subcollection targets (i.e. bulk requests for custom_attributes, tags, etc), above is needed to support multiple decrypts on a single automate workspace resource.  Maybe the name "resources" is misleading  and we need something else (i.e. attribute_references, etc).

Reply:
@abellotti @imtayadeway 
Do we need to support bulk decrypts? I don't think it makes much sense since we are doing this on a single method by method basis. Each Automate method might be doing something specific e.g
* Getting IP Address from IPAM system
* Getting some objects from LDAP system

They would typically be defined in 2 separate methods.

So can we not do bulk decrypts? and keep it simple by doing single decrypts (one attribute at a time)

Reply:
Ok by me.   We can always revisit later if it becomes an issue and we see many API round trips for this. Thanks.